osf/metrics/reporters/_base.py

```diff
@@ -20,7 +20,7 @@ def report(
     def run_and_record_for_month(self, report_yearmonth: YearMonth) -> None:
         reports = self.report(report_yearmonth)
         for report in reports:
-            assert report.report_yearmonth == str(report_yearmonth)
+            assert str(report.report_yearmonth) == str(report_yearmonth)
```

Comment:
Minor: The other MonthlyReporter casts `report_yearmonth` to `str` on `report` initialization -- should this one as well, instead of changing this assertion? It may not matter.

Reply:
hmm agree this is odd -- now looking at it again, seems better to set `report.report_yearmonth` here, instead of making the reporter do it (have updated)